src/Octoshift/GithubClient.cs

```diff
@@ -186,8 +204,24 @@ private string GetNextUrl(KeyValuePair<string, IEnumerable<string>>[] headers)
         private string ExtractLinkHeader(IEnumerable<KeyValuePair<string, IEnumerable<string>>> headers) =>
             ExtractHeaderValue("Link", headers);
 
+        private int RateLimitRemaining(IEnumerable<KeyValuePair<string, IEnumerable<string>>> headers) =>
+            int.Parse(ExtractHeaderValue("X-RateLimit-Remaining", headers) ?? "5000");
+
+        private long RateLimitReset(IEnumerable<KeyValuePair<string, IEnumerable<string>>> headers) =>
+            long.Parse(ExtractHeaderValue("X-RateLimit-Reset", headers) ?? DateTimeOffset.Now.ToUnixTimeSeconds().ToString());
```

Comment:
I would use the `_dateTimeProvider` abstraction:
```suggestion
            long.Parse(ExtractHeaderValue("X-RateLimit-Reset", headers) ?? _dateTimeProvider.CurrentUnixTimeSeconds().ToString());
```